common/src/main/java/org/apache/falcon/entity/FeedHelper.java

```diff
@@ -818,9 +819,11 @@ public static FeedInstanceResult getFeedInstanceListing(Entity entityObject,
                 instance.size = feedStatus.getSize();
                 instance.sizeH = feedStatus.getSizeH();
                 instances[index++] = instance;
+                allInstances.add(instance);
```

Comment:
It looks like that instances[index++] and allInstances.add() both are adding the feed instance. I think instances[] could have been used instead of creating new List variable allInstances by moving FeedInstanceResult.Instance[] instances above clusters for loop.